dataavailability/datacommittee/datacommittee.go

```diff
@@ -0,0 +1,292 @@
+	// request
+	c := client.New(member.URL)
+	log.Infof("sending request to sign the sequence to %s at %s", member.Addr.Hex(), member.URL)
+	signature, err := c.SignSequence(signedSequence)
+	if err != nil {
+		ch <- signatureMsg{
+			addr: member.Addr,
+			err:  err,
+		}
+		return
+	}
+	// verify returned signature
+	signedSequence.Signature = signature
+	signer, err := signedSequence.Signer()
+	if err != nil {
+		ch <- signatureMsg{
+			addr: member.Addr,
+			err:  err,
+		}
+		return
+	}
+	if signer != member.Addr {
+		ch <- signatureMsg{
+			addr: member.Addr,
+			err:  fmt.Errorf("invalid signer. Expected %s, actual %s", member.Addr.Hex(), signer.Hex()),
+		}
+		return
+	}
+	ch <- signatureMsg{
+		addr:      member.Addr,
+		signature: signature,
+	}
+}
+
+func buildSignaturesAndAddrs(msgs signatureMsgs, members []DataCommitteeMember) []byte {
+	res := []byte{}
+	sort.Sort(msgs)
+	for _, msg := range msgs {
+		log.Debugf("adding signature %s from %s", common.Bytes2Hex(msg.signature), msg.addr.Hex())
+		res = append(res, msg.signature...)
+	}
+	for _, member := range members {
+		log.Debugf("adding addr %s", common.Bytes2Hex(member.Addr.Bytes()))
+		res = append(res, member.Addr.Bytes()...)
+	}
+	log.Debugf("full res %s", common.Bytes2Hex(res))
+	return res
+}
+
+type signatureMsgs []signatureMsg
+
+func (s signatureMsgs) Len() int { return len(s) }
+func (s signatureMsgs) Less(i, j int) bool {
+	return strings.ToUpper(s[i].addr.Hex()) < strings.ToUpper(s[j].addr.Hex())
+}
+func (s signatureMsgs) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+// getCurrentDataCommittee return the currently registered data committee
+func (d *DataCommitteeBackend) getCurrentDataCommittee() (*DataCommittee, error) {
+	addrsHash, err := d.dataCommitteeContract.CommitteeHash(&bind.CallOpts{Pending: false})
+	if err != nil {
+		return nil, fmt.Errorf("error getting CommitteeHash from L1 SC: %w", err)
+	}
+	reqSign, err := d.dataCommitteeContract.RequiredAmountOfSignatures(&bind.CallOpts{Pending: false})
+	if err != nil {
+		return nil, fmt.Errorf("error getting RequiredAmountOfSignatures from L1 SC: %w", err)
+	}
+	members, err := d.getCurrentDataCommitteeMembers()
+	if err != nil {
+		return nil, err
+	}
+
+	return &DataCommittee{
+		AddressesHash:      common.Hash(addrsHash),
+		RequiredSignatures: reqSign.Uint64(),
+		Members:            members,
+	}, nil
+}
+
+// getCurrentDataCommitteeMembers return the currently registered data committee members
+func (d *DataCommitteeBackend) getCurrentDataCommitteeMembers() ([]DataCommitteeMember, error) {
+	members := []DataCommitteeMember{}
+	nMembers, err := d.dataCommitteeContract.GetAmountOfMembers(&bind.CallOpts{Pending: false})
+	if err != nil {
+		return nil, fmt.Errorf("error getting GetAmountOfMembers from L1 SC: %w", err)
+	}
+	for i := int64(0); i < nMembers.Int64(); i++ {
```

Comment:
Minor optimization, providing initial capacity to the slice WDYT?

```suggestion
	nMembers, err := d.dataCommitteeContract.GetAmountOfMembers(&bind.CallOpts{Pending: false})
	if err != nil {
		return nil, fmt.Errorf("error getting GetAmountOfMembers from L1 SC: %w", err)
	}
	members := make([]DataCommitteeMember, 0, nMembers.Int64())
```